File: src/dataset/cpp/beer-song_exercism-toemm/claude/ClaudeSonnet4_beer_song_v3.cpp
#include "beer_song.h"

#include <string>

namespace beer_song {

std::string verse(int i) {
    if (i == 0) {
        return "No more bottles of beer on the wall, no more bottles of beer.\n"
               "Go to the store and buy some more, 99 bottles of beer on the wall.\n";
    }

    const char* bottle_word = (i == 1) ? " bottle" : " bottles";
    const char* take_word = (i == 1) ? "it" : "one";
    
    std::string result;
    result.reserve(120);
    
    result += std::to_string(i);
    result += bottle_word;
    result += " of beer on the wall, ";
    result += std::to_string(i);
    result += bottle_word;
    result += " of beer.\nTake ";
    result += take_word;
    result += " down and pass it around, ";
    
    if (i == 1) {
        result += "no more bottles";
    } else {
        result += std::to_string(i - 1);
        result += (i == 2) ? " bottle" : " bottles";
    }
    
    result += " of beer on the wall.\n";
    
    return result;
}

std::string sing(int begin, int ending) {
    std::string result;
    const int verse_count = begin - ending + 1;
    result.reserve(verse_count * 120);
    
    for (int i = begin; i >= ending; --i) {
        result += verse(i);
        if (i != ending) {
            result += '\n';
        }
    }
    
    return result;
}

}  // namespace beer_song

File: src/dataset/javascript/beer-song_exercism-elenamountz/claude/ClaudeSonnet4_beer-song_v3.js
export default class Beer {
  constructor() {}

  static verse(n) {
    const bottleWord = n === 1 ? 'bottle' : 'bottles';
    
    if (n === 0) {
      return 'No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, 99 bottles of beer on the wall.\n';
    }
    
    if (n === 1) {
      return '1 bottle of beer on the wall, 1 bottle of beer.\nTake it down and pass it around, no more bottles of beer on the wall.\n';
    }
    
    const nextBottleWord = n === 2 ? 'bottle' : 'bottles';
    return `${n} ${bottleWord} of beer on the wall, ${n} ${bottleWord} of beer.\nTake one down and pass it around, ${n - 1} ${nextBottleWord} of beer on the wall.\n`;
  }

  static sing(from = 99, to = 0) {
    const verses = [];
    for (let i = from; i >= to; i--) {
      verses.push(this.verse(i));
    }
    return verses.join('\n').slice(0, -1);
  }
}